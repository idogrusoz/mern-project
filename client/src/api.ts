import axios from "axios";

export default axios.create({
    baseURL: process.env.NODE_ENV === "production" ? process.env.PUBLIC_URL : `http://localhost:5000/api/v1/`,
    withCredentials: true,
});
