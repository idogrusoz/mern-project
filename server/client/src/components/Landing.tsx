import { Container } from "@material-ui/core";
import React, { useContext } from "react";
import { AuthContext } from "./auth/AuthContext";
import Register from "./auth/Register";
import SignIn from "./auth/SignIn";

const Landing = () => {
    const { showSignin } = useContext(AuthContext);
    return <Container maxWidth="sm">{showSignin ? <SignIn /> : <Register />}</Container>;
};

export default Landing;
