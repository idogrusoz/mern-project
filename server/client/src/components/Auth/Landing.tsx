import { Container, Typography } from "@material-ui/core";
import React, { useCallback, useContext, useState } from "react";
import { SignInUser, UserInterface } from "../../../../interfaces/user.interfaces";
import api from "../../api";
import { AuthContext } from "./AuthContext";
import Register from "./Register";
import SignIn from "./SignIn";

const Landing = () => {
    const [showSignin, setShowSignin] = useState<boolean>(true);
    const [error, setError] = useState<boolean>(false);
    const [errorMessage, setErrorMessage] = useState<string>("");
    const { setAuthenticated, setUser } = useContext(AuthContext);

    const register = useCallback(async (user: UserInterface) => {
        try {
            const response = await api.post("auth/register", user);
            setShowSignin(response.status === 200);
        } catch (error) {
            setError(true);
            setErrorMessage(error.message);
        }
    }, []);

    const signin = useCallback(
        async (user: SignInUser) => {
            try {
                const response = await api.post("auth/signin", user);
                setAuthenticated(response.data.isAuthenticated);
                setUser(response.data.user);
            } catch (error) {
                setError(true);
                setErrorMessage(error.message);
            }
        },
        [setAuthenticated, setUser],
    );

    return (
        <Container maxWidth="sm">
            {error && <Typography color="secondary">{errorMessage}</Typography>}
            {showSignin ? (
                <SignIn signin={signin} setShowSignin={setShowSignin} />
            ) : (
                <Register register={register} setShowSignin={setShowSignin} />
            )}
        </Container>
    );
};

export default Landing;
