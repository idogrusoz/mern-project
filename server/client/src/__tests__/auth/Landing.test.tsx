jest.mock("../../api.ts");
import { Button, TextField } from "@material-ui/core";
import { act } from "@testing-library/react";
import { mount, ReactWrapper } from "enzyme";
import React from "react";
import api from "../../api";
import Landing from "../../components/auth/Landing";

describe("Landing component", () => {
    let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
    beforeEach(() => {
        wrapper = mount(<Landing />);
    });
    it("renders without crashing", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("renders switches between signin ad register", () => {
        expect(wrapper.find(TextField)).toHaveLength(2);
        wrapper.find(Button).at(1).simulate("click");
        expect(wrapper.find(TextField)).toHaveLength(5);
    });
    it("signs in the user", async () => {
        const button = wrapper.find(Button).first();
        expect(button.text()).toBe("Sign In");
        const click = async () => {
            await button.simulate("click");
        };
        await act(click);
        expect(api.post).toBeCalled();
    });
    it("signs in the user", async () => {
        const button = wrapper.find(Button).at(1);
        button.simulate("click");
        const registerButton = wrapper.find(Button).first();
        expect(registerButton.text()).toBe("Register");
        const click = async () => {
            await registerButton.simulate("click");
        };
        await act(click);
        expect(api.post).toBeCalled();
    });
});
