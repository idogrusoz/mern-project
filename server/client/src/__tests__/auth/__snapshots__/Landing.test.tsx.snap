// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders without crashing 1`] = `ReactWrapper {}`;
